File: scrivabesrv/src/java/it/csi/scriva/scrivabesrv/business/be/service/AmbitiService.java
/*-
 * ========================LICENSE_START=================================
 * 
 * Copyright (C) 2025 Regione Piemonte
 * 
 * SPDX-FileCopyrightText: (C) Copyright 2025 Regione Piemonte
 * SPDX-License-Identifier: EUPL-1.2
 * =========================LICENSE_END==================================
 */
package it.csi.scriva.scrivabesrv.business.be.service;

import it.csi.scriva.scrivabesrv.dto.AmbitoDTO;

import java.util.List;

/**
 * The interface Ambiti service.
 *
 * @author CSI PIEMONTE
 */
public interface AmbitiService {

    /**
     * Load ambito list.
     *
     * @param idAmbito  the id ambito
     * @param codAmbito the cod ambito
     * @return the list
     */
    List<AmbitoDTO> loadAmbiti(Long idAmbito, String codAmbito);

    /**
     * Load json ambiti string.
     *
     * @param idAmbito  the id ambito
     * @param codAmbito the cod ambito
     * @return the string
     */
    String loadJsonAmbiti(Long idAmbito, String codAmbito);

    /**
     * Load ambito list.
     *
     * @param idAmbito the id ambito
     * @return the list
     */
    List<AmbitoDTO> loadAmbito(Long idAmbito);

}

File: scrivafowcl/src/app/shared/services/ambiti.service.ts
/*
 * ========================LICENSE_START=================================
 * 
 * Copyright (C) 2025 Regione Piemonte
 * 
 * SPDX-FileCopyrightText: (C) Copyright 2025 Regione Piemonte
 * SPDX-License-Identifier: EUPL-1.2
 * =========================LICENSE_END==================================
 */
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { map, tap } from 'rxjs/operators';

import { AppConfigService } from './app-config.service';
import { Ambito } from '../models';

@Injectable()
export class AmbitiService {
  private beUrl = '';

  ambiti: Ambito[] = [];

  constructor(private http: HttpClient, private config: AppConfigService) {
    this.beUrl = this.config.getBEUrl();
  }

  /*
   * API ambiti
   */
  getAmbiti(): Observable<Ambito[]> {
    return this.http.get<Ambito[]>(`${this.beUrl}/ambiti`).pipe(
      tap(res => this.ambiti = res),
    );
  }

  getAmbitoById(id: number): Observable<Ambito> {
    return this.http
      .get<Ambito[]>(`${this.beUrl}/ambiti/${id}`)
      .pipe(
        map((res) => res[0])
      );
  }

  getAmbitoByCode(codAmbito: string): Observable<Ambito> {
    return this.http
      .get<Ambito[]>(`${this.beUrl}/ambiti?codice=${codAmbito}`)
      .pipe(
        map((res) => res[0])
      );
  }
}
